src/Tmds.LinuxAsync/IOUringAsyncEngine.IOUringExecutionQueue.cs

```diff
@@ -258,8 +291,10 @@ public void ExecuteCompletions()
                 while (_ring!.TryRead(out Completion completion))
                 {
                     ulong key = completion.userData;
-                    if (_operations.Remove(key, out Operation? op))
+                    if ((key & PollMaskBit) == 0)
```

Comment:
nit: maybe `if ((key & PollMaskBit) != 0)` so the cases are in 'chronological' order.